com.unity.render-pipelines.core/Runtime/RenderGraph/RenderGraphResourceRegistry.cs

```diff
@@ -595,65 +586,42 @@ void SetGlobalTextures(RenderGraphContext rgContext, List<TextureHandle> texture
         }
 
-        internal void PostRenderPassUnbindGlobalTextures(RenderGraphContext rgContext, List<TextureHandle> textures)
+        internal void PostRenderPassUnbindGlobalTextures(RenderGraphContext rgContext, IReadOnlyCollection<TextureHandle> textures)
         {
             SetGlobalTextures(rgContext, textures, true);
         }
 
-        internal void ReleaseTexturesForPass(RenderGraphContext rgContext, int passIndex, List<TextureHandle> readTextures, List<TextureHandle> writtenTextures)
+        internal void ReleaseTexture(RenderGraphContext rgContext, TextureHandle resource)
         {
-            foreach (var resource in readTextures)
+            ref var resourceDesc = ref GetTextureResource(resource);
+            if (!resourceDesc.imported)
             {
-                ref var resourceDesc = ref GetTextureResource(resource);
-                if (!resourceDesc.imported && resourceDesc.lastReadPassIndex == passIndex)
+                if (m_RenderGraphDebug.clearRenderTargetsAtRelease)
                 {
-                    if (m_RenderGraphDebug.clearRenderTargetsAtRelease)
+                    using (new ProfilingScope(rgContext.cmd, ProfilingSampler.Get(RenderGraphProfileId.RenderGraphClearDebug)))
                     {
-                        using (new ProfilingScope(rgContext.cmd, ProfilingSampler.Get(RenderGraphProfileId.RenderGraphClearDebug)))
-                        {
-                            var clearFlag = resourceDesc.desc.depthBufferBits != DepthBits.None ? ClearFlag.Depth : ClearFlag.Color;
-                            CoreUtils.SetRenderTarget(rgContext.cmd, GetTexture(resource), clearFlag, Color.magenta);
-                        }
+                        var clearFlag = resourceDesc.desc.depthBufferBits != DepthBits.None ? ClearFlag.Depth : ClearFlag.Color;
+                        CoreUtils.SetRenderTarget(rgContext.cmd, GetTexture(resource), clearFlag, Color.magenta);
                     }
-
-                    ReleaseTextureForPass(resource);
                 }
-            }
 
-            // If a resource was created for only a single pass, we don't want users to have to declare explicitly the read operation.
-            // So to do that, we also update lastReadIndex on resource writes.
-            // This means that we need to check written resources for destruction too
-            foreach (var resource in writtenTextures)
-            {
-                ref var resourceDesc = ref GetTextureResource(resource);
-                // <= because a texture that is only declared as written in a single pass (and read implicitly in the same pass) will have the default lastReadPassIndex at -1
-                if (!resourceDesc.imported && resourceDesc.lastReadPassIndex <= passIndex)
+                //// This can happen because we release texture in two passes (see ReleaseTexturesForPass) and texture can be present in both passes
+                //if (resourceDesc.rt != null)
```

Comment:
expect this comment here ?

Reply:
Nope, will remove the code. The test is not necessary anymore.